SeaPublicWebsite/Controllers/EnergyEfficiencyController.cs

```diff
@@ -249,72 +244,98 @@ public async Task<IActionResult> AskForPostcode_Post(AskForPostcodeViewModel vie
             {
                 return await AskForPostcode_Get(viewModel.Reference);
             }
-            List<EpcInformation> epcInformationList = await epcApi.GetEpcsInformationForPostcodeAndBuildingNameOrNumber(viewModel.Postcode, viewModel.HouseNameOrNumber);
+            
+            var propertyData = await propertyDataStore.LoadPropertyDataAsync(viewModel.Reference);
+            var forwardArgs = questionFlowService.ForwardLinkArguments(QuestionFlowPage.AskForPostcode, propertyData);
 
-            return await UpdatePropertyAndRedirect(
-                p =>
-                {
-                    p.Postcode = viewModel.Postcode;
-                    p.HouseNameOrNumber = viewModel.HouseNameOrNumber;
-                    p.EpcCount = epcInformationList.Count;
-                    p.Epc = null;
-                    p.EpcDetailsConfirmed = null;
-                    p.EpcAddressConfirmed = EpcAddressConfirmed.No;
-                },
-                viewModel.Reference,
-                QuestionFlowPage.AskForPostcode);
+            return RedirectToAction(
+                forwardArgs.Action,
+                forwardArgs.Controller,
+                // TODO: seabeta-579 we'd rather use the values from the question flow service
+                new { reference = viewModel.Reference, postcode = viewModel.Postcode, number = viewModel.HouseNameOrNumber });
         }
 
         
-        [HttpGet("address/{reference}")]
-        public async Task<ViewResult> ConfirmAddress_Get(string reference)
+        [HttpGet("address/{reference}/{postcode}/{number}")]
+        public async Task<ViewResult> ConfirmAddress_Get(string reference, string postcode, string number)
```

Comment:
any reason for renaming this from `houseNameOrNumber` to `number`?

Reply:
Not really, it was just shorter